tests/dashboard/plot_orchestrator_test.py

```diff
@@ -1126,5 +1189,534 @@ def test_add_cell_immediately_remove_it_then_workflow_commits_no_error(
+            )
+            fake_data_service[result_key] = sc.scalar(1.0)
+
+        # Simulate late subscriber (new session, page reload, etc.)
+        # Late subscriber retrieves grid config
+        grids = plot_orchestrator.get_all_grids()
+        assert grid_id in grids
+
+        grid_config = grids[grid_id]
+        assert len(grid_config.cells) == 3
+
+        # Late subscriber retrieves plots for all cells via get_cell_state
+        for cell_id in cell_ids:
+            plot, error = plot_orchestrator.get_cell_state(cell_id)
+            assert plot is not None, f"Plot should exist for cell {cell_id}"
+            assert isinstance(plot, FakePlot)
+            assert error is None
+
+    def test_get_cell_state_updated_when_cell_config_updated_with_running_workflow(
+        self,
+        plot_orchestrator,
+        plot_cell,
+        workflow_id,
+        workflow_spec,
+        job_orchestrator,
+        fake_data_service,
+    ):
+        """
+        When config is updated and workflow is running, plot is immediately recreated.
+        """
+        grid_id = plot_orchestrator.add_grid(title='Test Grid', nrows=3, ncols=3)
+        cell_id = plot_orchestrator.add_plot(grid_id, plot_cell)
+
+        # Commit workflow
+        job_ids = commit_workflow_for_test(job_orchestrator, workflow_id, workflow_spec)
+        job_number1 = job_ids[0].job_number
+
+        # Simulate data arrival for ALL sources
+        import scipp as sc
+
+        from ess.livedata.config.workflow_spec import JobId, ResultKey
+
+        for source_name in plot_cell.config.source_names:
+            result_key = ResultKey(
+                workflow_id=workflow_id,
+                job_id=JobId(source_name=source_name, job_number=job_number1),
+                output_name=plot_cell.config.output_name,
+            )
+            fake_data_service[result_key] = sc.scalar(1.0)
+
+        # Verify plot exists
+        plot1, error = plot_orchestrator.get_cell_state(cell_id)
+        assert plot1 is not None
+        assert error is None
+
+        # Update config while workflow is still running
+        new_config = PlotConfig(
+            workflow_id=workflow_id,
+            output_name='new_output',
+            source_names=['new_source'],
+            plot_name='new_plot',
+            params={},
+        )
+        plot_orchestrator.update_plot_config(cell_id, new_config)
+
+        # Simulate data arrival for new config
+        result_key2 = ResultKey(
+            workflow_id=workflow_id,
+            job_id=JobId(source_name='new_source', job_number=job_number1),
+            output_name='new_output',
+        )
+        fake_data_service[result_key2] = sc.scalar(2.0)
+
+        # Since workflow is still running, plot should be immediately recreated
+        plot2, error = plot_orchestrator.get_cell_state(cell_id)
+        assert plot2 is not None
+        assert error is None
+        # Note: We can't reliably test for a different instance since
```

Comment:
Is it possible to check something about the config or the result to check the cell_id is now pointing to the new data?

Reply:
Done!